lib/shopify_cli/git.rb

```diff
@@ -63,18 +63,16 @@ def sha(dir: Dir.pwd, ctx: Context.new)
       #
       #   ShopifyCLI::Git.clone('git@github.com:shopify/test.git', 'test-app')
       #
-      def clone(repository, dest, ctx: Context.new)
+      def clone(repository, dest, ctx: Context.new, &block)
```

Comment:
I believe this PR is in a good direction already!

An alternative approach would be:
- using a NullObject for the frame logic, and
- having a new `raw_clone` method (mirroring `clone`) - at the `raw_clone` we'd received the block, and at the regular `clone` we'd use the `CLI::UI::Progress.progress`

That alternative would remove the `if block_given?` from the line `204`.

Just sharing those thoughts, because maybe they valuable during some refactoring at testing phase. But, I personally would keep the PR as is :)